src/main/java/ttt/Board.java

```diff
@@ -1,61 +1,89 @@
 package ttt;
 
 import java.util.Arrays;
+import java.util.function.Function;
 
 import static ttt.PlayerSymbol.*;
-import static ttt.PlayerSymbol.VACANT;
 
 public class Board {
-    private PlayerSymbol[] grid = new PlayerSymbol[BOARD_DIMENSION * BOARD_DIMENSION];
-
     protected static final int BOARD_DIMENSION = 3;
     private static final int NUMBER_OF_SLOTS = BOARD_DIMENSION * BOARD_DIMENSION;
+    private Cell[] grid = new Cell[BOARD_DIMENSION * BOARD_DIMENSION];
 
     public Board() {
-        for (int i = 0; i < NUMBER_OF_SLOTS; i++) {
-            grid[i] = VACANT;
+        for (int cellIndex = 0; cellIndex < NUMBER_OF_SLOTS; cellIndex++) {
+            grid[cellIndex] = new Cell(calculateOffsetFor(cellIndex), VACANT);
         }
     }
 
     public Board(PlayerSymbol... grid) {
-        this.grid = grid;
+        for (int cellIndex = 0; cellIndex < NUMBER_OF_SLOTS; cellIndex++) {
+            this.grid[cellIndex] = new Cell(calculateOffsetFor(cellIndex), grid[cellIndex]);
```

Comment:
`this.grid` and `grid` in the same method is confusing as the name is shared amongst different things.

The `PlayerSymbol` objects you are passing in and calling `grid` refer to the initial layout you want the `grid` to look like when you setup the game. Is that a right assumption? If so `initialGridLayout` would be a better name choice.